pyriemann/utils/ajd.py

```diff
@@ -269,19 +279,19 @@ def uwedge(X, init=None, eps=1e-7, n_iter_max=100):
 
     crit = np.sum(Ms**2) - np.sum(Rs**2)
     while (improve > eps) & (iteration < n_iter_max):
-        B = np.dot(Rs, Rs.T)
+        B = Rs @ Rs.T
         C1 = np.zeros((d, d))
         for i in range(d):
             C1[:, i] = np.sum(Ms[:, i:Md:d]*Rs, axis=1)
 
-        D0 = B*B.T - np.outer(np.diag(B), np.diag(B))
-        A0 = (C1 * B - np.dot(np.diag(np.diag(B)), C1.T)) / (D0 + np.eye(d))
+        D0 = B * B.T - np.outer(np.diag(B), np.diag(B))
+        A0 = (C1 * B - np.diag(np.diag(B)) @ C1.T) / (D0 + np.eye(d))
         A0 += np.eye(d)
         W_est = np.linalg.solve(A0, W_est)
 
         Raux = np.dot(np.dot(W_est, M[:, 0:d]), W_est.T)
-        aux = 1./np.sqrt(np.abs(np.diag(Raux)))
-        W_est = np.dot(np.diag(aux), W_est)
+        aux = 1. / np.sqrt(np.abs(np.diag(Raux)))
+        W_est = np.diag(aux) @ W_est
```

Comment:
use broadcasting if you can. Every time I see a np.diag in numpy my brain produces a strong ERP ;)

Reply:
I agree. I try to remove np.dot and np.diag. But I only improve the code when I’m sure to break nothing.

Reply:
uwedge seems deep, I couldn't think of a way to get rid of the np.diag nicely.